python/_tskitmodule.c

```diff
@@ -6256,6 +6256,38 @@ TableCollection_link_ancestors(TableCollection *self, PyObject *args, PyObject *
     return ret;
 }
 
+static PyObject *
+TableCollection_subset(TableCollection *self, PyObject *args)
+{
+    int err;
+    PyObject *ret = NULL;
+    PyObject *nodes = NULL;
+    PyArrayObject *nodes_array = NULL;
+    npy_intp *shape;
+    size_t num_nodes;
+
+    if (!PyArg_ParseTuple(args, "O", &nodes)) {
+        goto out;
+    }
+    nodes_array = (PyArrayObject *) PyArray_FROMANY(nodes, NPY_INT32, 1, 1,
+            NPY_ARRAY_IN_ARRAY);
+    if (nodes_array == NULL) {
+        goto out;
+    }
+    shape = PyArray_DIMS(nodes_array);
+    num_nodes = shape[0];
+
+    err = tsk_table_collection_subset(self->tables, PyArray_DATA(nodes_array), num_nodes);
+    if (err != 0) {
+        handle_library_error(err);
+        goto out;
+    }
+    ret = Py_BuildValue("");
+out:
```

Comment:
Very nice, this is perfect. We need some tests of this in test_lowlevel.py though, to make sure that the error conditions are handled correctly. We'll want to call the new method with, e.g., the incorrect number of arguments and bad types for numpy to make sure these error conditions are handled.